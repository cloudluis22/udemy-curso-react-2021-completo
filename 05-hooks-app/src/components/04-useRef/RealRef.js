import React, { useState } from 'react';
import '../02-useEffect/effect.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealRef = () => {

    const [show, setShow] = useState(false);

    const handleComponentStatus = () => { setShow( !show ) };

    return (
        <div>
            <h1>Real Example Use of useRef</h1>
            <hr />

            { show && <MultipleCustomHooks />}

            <button className="btn btn-primary mt-5" onClick={handleComponentStatus}> Show/Hide </button>
        </div>
    )
}
