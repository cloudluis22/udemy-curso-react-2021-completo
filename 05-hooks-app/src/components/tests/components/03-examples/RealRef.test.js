import React from 'react';
import { shallow } from 'enzyme';
import { RealRef } from '../../../04-useRef/RealRef';

describe('Pruebas sobre el componente de <RealRef />.', () => {

    const wrapper = shallow(<RealRef />);

    test('Debe de mostrar el componente correctamente y crear una snapshot.', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar el componente de <MultipleCustomHooks /> una vez se presiona el botÃ³n.', () => {


    });

});
