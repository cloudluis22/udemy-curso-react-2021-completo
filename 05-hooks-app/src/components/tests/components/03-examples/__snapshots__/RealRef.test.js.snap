// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente de <RealRef />. Debe de mostrar el componente correctamente y crear una snapshot. 1`] = `
<div>
  <h1>
    Real Example Use of useRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
     Show/Hide 
  </button>
</div>
`;
